lib/commands.js

```diff
@@ -17,18 +17,14 @@ var changeFunctionName = (function () {
     }
 }());
 
-// TODO: Rewrite this including the invidual commands into a Commands class
-// that provided a functionality to add new commands to the client
-
-commands.list.forEach(function (command) {
-
+var addCommand = function (command) {
     // Some rare Redis commands use special characters in their command name
     // Convert those to a underscore to prevent using invalid function names
     var commandName = command.replace(/(?:^([0-9])|[^a-zA-Z0-9_$])/g, '_$1');
 
     // Do not override existing functions
-    if (!RedisClient.prototype[command]) {
-        RedisClient.prototype[command.toUpperCase()] = RedisClient.prototype[command] = function () {
+    if (!RedisClient.prototype[commandName]) {
+        RedisClient.prototype[commandName.toUpperCase()] = RedisClient.prototype[commandName] = function () {
```

Comment:
Ok - this bit of code tripped me up. I think the way it is currently implemented isn't correct. In the current master the `commands.js` has the following bits (save for the `console.log` I inserted and noted below):
```
    // Some rare Redis commands use special characters in their command name
    // Convert those to a underscore to prevent using invalid function names
    var commandName = command.replace(/(?:^([0-9])|[^a-zA-Z0-9_$])/g, '_$1');
    console.log(command); //added to show all commands
    // Do not override existing functions
    if (!RedisClient.prototype[command]) { ...
```
The will show all the commands as they are added to the client - including two with strange characters (`host:` and `restore-asking`) which yields two functions with illegal characters `RedisClient.prototype['host:']` and `RedisClient.prototype['restore-asking']`. So shouldn't it actually be `RedisClient.prototype[commandName]` to yield `RedisClient.prototype.host_` and `RedisClient.prototype['restore_asking']`?